src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,1441 @@
+    } else if (visualizer.pathType == PathType.Rows) {
+        currentDirection = 'right'
+    }
+}
+
+function changeDirectionBasedOnPathType(
+    visualizer: VisualizerGrid,
+    squareRootOfAmountOfNumbers: number,
+    amountOfNumbers: number,
+    iteration: number
+) {
+    //Choose direction for next number
+    if (visualizer.pathType == PathType.Spiral) {
+        //Turn at the numberToTurn at which increases every other turn
+        if (iteration == numberToTurnAt) {
+            numberToTurnAt += incrementForNumberToTurnAt
+            if (whetherIncrementShouldIncrementForSpiral) {
+                incrementForNumberToTurnAt += 1
+                whetherIncrementShouldIncrementForSpiral = false
+            } else {
+                whetherIncrementShouldIncrementForSpiral = true
+            }
+
+            if (currentDirection == 'right') {
+                currentDirection = 'up'
+            } else if (currentDirection == 'up') {
+                currentDirection = 'left'
+            } else if (currentDirection == 'left') {
+                currentDirection = 'down'
+            } else if (currentDirection == 'down') {
+                currentDirection = 'right'
+            }
+        }
+    } else if (visualizer.pathType == PathType.Inwards_Spiral) {
+        //Turn at the numbers to turn at which decreases every other turn
+        if (iteration == numberToTurnAt) {
+            numberToTurnAt += incrementForNumberToTurnAt
+            if (whetherIncrementShouldIncrementForSpiral) {
+                incrementForNumberToTurnAt -= 1
+                whetherIncrementShouldIncrementForSpiral = false
+            } else {
+                whetherIncrementShouldIncrementForSpiral = true
+            }
+
+            if (currentDirection == 'right') {
+                currentDirection = 'down'
+            } else if (currentDirection == 'down') {
+                currentDirection = 'left'
+            } else if (currentDirection == 'left') {
+                currentDirection = 'up'
+            } else if (currentDirection == 'up') {
+                currentDirection = 'right'
+            }
+        }
+    } else if (visualizer.pathType == PathType.Rows) {
+        //Go to new row when the row is complete
+        if ((iteration + 1) % squareRootOfAmountOfNumbers == 0) {
+            currentDirection = 'startNewRow'
+        } else if (iteration == amountOfNumbers) {
+            currentDirection = 'none'
+        } else {
+            currentDirection = 'right'
+        }
+    }
+}
+
+function showPath(
+    visualizer: VisualizerGrid,
+    iteration: number,
+    previousDirection: string,
+    currentDirection: string,
+    scalingFactor: number
+) {
+    if (visualizer.visualInfo == VisualInfo.Show_Path) {
+        visualizer.sketch.fill(visualizer.pathAndNumberColor)
+
+        if (iteration != 0) {
+            if (previousDirection == 'left') {
+                visualizer.sketch.rect(
```

Comment:
OK, in this function is one example of something that was discussed in the recent coding practices session: a big conditional with very similar code blocks in each case. Could you use something like any of the options we discussed to align it better with KISS and DRY?

Reply:
I guess you "solved" this by completely eliminating the functionality, correct? Do you see any other places in your code where a similar observation might be made, though?

Reply:
I've tried to eliminate other repetition I notice.

Reply:
Will be revisited in further review, I imagine.